tests/manage/mcg/test_object_integrity.py

```diff
@@ -0,0 +1,90 @@
+import logging
+
+import boto3
+import pytest
+
+from ocs_ci.framework import config
+from ocs_ci.framework.testlib import ManageTest, tier1
+from ocs_ci.ocs import constants
+from tests.helpers import craft_s3_command
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.skipif(
+    condition=config.ENV_DATA['platform'] != 'AWS',
+    reason="Tests are not running on AWS deployed cluster"
+)
+def check_md5sum(original_object, result_object, awscli_pod):
+    md5sum = awscli_pod.exec_cmd_on_pod(
+        command=f'md5sum {original_object} {result_object}'
+    )
+    md5sum_original = md5sum.split()[0]
+    md5sum_result = md5sum.split()[2]
+    assert md5sum_original == md5sum_result, (
+        'Data Corruption Found'
+    )
+    logger.info(f'Passed: MD5 comparison for '
+                f'{original_object} and {result_object}')
+    return True
```

Comment:
We need alignment on the returned value.
or we are raising an exception (Line 24) or we returning a bool.